### ISSUE TITLE
TypeError: Return value command

### ISSUE THREAD

**Issue by gsantner** [Rello/audioplayer#540]
**Describe the bug**

```
[gregor@gregor-laptop nextcloud]$ sudo -u www-data php ./occ --version
Nextcloud 20.0.10

[gregor@gregor-laptop nextcloud]$ find ~/Musik/mp3/ -iname "*.mp3" | wc -l
2191

[gregor@gregor-laptop nextcloud]$ cat apps/audioplayer/appinfo/info.xml | grep -A3 version
<?xml version="1.0"?>
<info xmlns:xsi= "http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="https://apps.nextcloud.com/schema/apps/info.xsd">
	<id>audioplayer</id>
	<name>Audio Player</name>
--
	<version>3.1.0</version>
	<licence>agpl</licence>
	<author>Marcel Scherello</author>
	<namespace>audioplayer</namespace>
--
		<nextcloud min-version="20" max-version="21"/>
	</dependencies>


[gregor@gregor-laptop nextcloud]$ sudo -u www-data php ./occ audioplayer:scan --all
[sudo] Passwort für gregor: 
Start scan for admin
Start processing of audio files
Audios found: 806
Duplicates found: 0
Written to library: 799
Albums found: 143
Errors: 7
Start scan for gregor
Start processing of audio files
Audios found: 23
Duplicates found: 0
Written to library: 16
Albums found: 144
Errors: 7
An unhandled exception has been thrown:
TypeError: Return value of "OCA\audioplayer\Command\Scan::execute()" must be of the type int, "null" returned. in /var/www/nextcloud/apps/mail/vendor/symfony/console/Command/Command.php:261
Stack trace:
#0 /var/www/nextcloud/apps/mail/vendor/symfony/console/Application.php(920): Symfony\Component\Console\Command\Command->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#1 /var/www/nextcloud/apps/mail/vendor/symfony/console/Application.php(266): Symfony\Component\Console\Application->doRunCommand(Object(OCA\audioplayer\Command\Scan), Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#2 /var/www/nextcloud/apps/mail/vendor/symfony/console/Application.php(142): Symfony\Component\Console\Application->doRun(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#3 /var/www/nextcloud/lib/private/Console/Application.php(215): Symfony\Component\Console\Application->run(Object(Symfony\Component\Console\Input\ArgvInput), Object(Symfony\Component\Console\Output\ConsoleOutput))
#4 /var/www/nextcloud/console.php(100): OC\Console\Application->run()
#5 /var/www/nextcloud/occ(11): require_once('/var/www...')
```

**To Reproduce**
1. Start scan, via occ or frontend
2. Crashes and not even 50% of music gets indexed
3. Not sure what is done with the mail addon here. Faulty import?

**Expected behavior**
Index all (valid) music (files) correctly.

**Screenshots**  
See paste 

**Versions:**
 - Nextcloud: See paste
 - Audioplayer See paste

**Comment by gsantner** [comment]
If I retry the command multiple times, it from time to time seems to add new audio files. Is it how it is supposed to work? Randomly index some files? Instead of "index all" as I would as user understand it?

```
[gregor@gregor-laptop nextcloud]$ sudo -u www-data php ./occ audioplayer:scan --all
Start scan for admin
Start processing of audio files
Audios found: 1013
Duplicates found: 0
Written to library: 1006
Albums found: 139
Errors: 7
Start scan for gregor
Start processing of audio files
Audios found: 1075
Duplicates found: 0
Written to library: 1068
Albums found: 287
Errors: 7
[gregor@gregor-laptop nextcloud]$ sudo -u www-data php ./occ audioplayer:scan --all
Start scan for admin
Start processing of audio files
Audios found: 458
Duplicates found: 0
Written to library: 450
Albums found: 124
Errors: 8
Start scan for gregor
Start processing of audio files
Audios found: 396
Duplicates found: 0
Written to library: 388
Albums found: 239
Errors: 8
[gregor@gregor-laptop nextcloud]$ sudo -u www-data php ./occ audioplayer:scan --all
Start scan for admin
Start processing of audio files
Audios found: 8
Duplicates found: 0
Written to library: 0
Albums found: 0
Errors: 8
Start scan for gregor
Start processing of audio files
Audios found: 8
Duplicates found: 0
Written to library: 0
Albums found: 0
Errors: 8
```

**Comment by gsantner** [comment]
When I disable the mail addon, the stacktrace above is gone. Is there some bad import?

If I retry the command multiple times then, audioplayer from time to time seems to add new audio files. Is it how it is supposed to work? Randomly index some files? Instead of "index all" as I would as user understand it?

```
[gregor@gregor-laptop nextcloud]$ sudo -u www-data php ./occ audioplayer:scan --all
Start scan for admin
Start processing of audio files
Audios found: 1013
Duplicates found: 0
Written to library: 1006
Albums found: 139
Errors: 7
Start scan for gregor
Start processing of audio files
Audios found: 1075
Duplicates found: 0
Written to library: 1068
Albums found: 287
Errors: 7
[gregor@gregor-laptop nextcloud]$ sudo -u www-data php ./occ audioplayer:scan --all
Start scan for admin
Start processing of audio files
Audios found: 458
Duplicates found: 0
Written to library: 450
Albums found: 124
Errors: 8
Start scan for gregor
Start processing of audio files
Audios found: 396
Duplicates found: 0
Written to library: 388
Albums found: 239
Errors: 8
[gregor@gregor-laptop nextcloud]$ sudo -u www-data php ./occ audioplayer:scan --all
Start scan for admin
Start processing of audio files
Audios found: 8
Duplicates found: 0
Written to library: 0
Albums found: 0
Errors: 8
Start scan for gregor
Start processing of audio files
Audios found: 8
Duplicates found: 0
Written to library: 0
Albums found: 0
Errors: 8
```

**Comment by Rello** [comment]
Hi,
in general, the scanner can be restarted after every error and it will pick up to scan the rest.
I don´t know if there are logs when the scanner stops, but this can be the case when e.g. a php-memory limit is hit.
it normal operation it should do all in one.

I will check for the return code error. its related to the change from NC19/NC20

**Comment by Rello** [comment]
occ scan errors when scan folder is not available anymore #539

**Comment by Rello** [comment]
issue is solved with the next release. sorry for the delay
